src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -322,32 +324,43 @@ private void OnStartingBackgroundIdleWork()
                 StopIdleTimer();
 
                 // We need to get back to the UI thread to properly check if a completion is active.
-                await Task.Factory.StartNew(OnIdle, null, CancellationToken.None, TaskCreationOptions.None, _dispatcher.ForegroundScheduler);
+                Task.Factory.StartNew(OnIdle, null, CancellationToken.None, TaskCreationOptions.None, _dispatcher.ForegroundScheduler);
             }
             catch (Exception ex)
             {
                 // This is something totally unexpected, let's just send it over to the workspace.
-                await Task.Factory.StartNew(() => _errorReporter.ReportError(ex), CancellationToken.None, TaskCreationOptions.None, _dispatcher.ForegroundScheduler);
+                Task.Factory.StartNew(() => _errorReporter.ReportError(ex), CancellationToken.None, TaskCreationOptions.None, _dispatcher.ForegroundScheduler);
             }
         }
 
         private void OnResultsReady(object sender, DocumentStructureChangedEventArgs args)
         {
             _dispatcher.AssertBackgroundThread();
 
-            if (DocumentStructureChanged != null)
-            {
-                if (args.Snapshot != TextBuffer.CurrentSnapshot)
-                {
-                    // A different text change is being parsed.
-                    return;
-                }
+            Interlocked.Increment(ref _pendingStructureChanges);
```

Comment:
Does this happen before or after a change to `_parser.IsIdle`?

Reply:
after